### ISSUE TITLE
from_pretrained

### ISSUE THREAD

**Issue by steindor** [huggingface/pytorch-transformers#631]
I have a question regarding the from_pretrained method since I experienced a bit unexpected behaviour. It's regarding how to save a BERT classifier model as a whole.

I am experimenting with a classifier on top of BERT on the stack overflow question / tags dataset which contains 20 classes of 40.000 text samples. I trained a classifier on 80% of the data for 3 epochs and saved the model in the following and recommended way:

```
def save_model(model):
    # Save a trained model, configuration and tokenizer
    model_to_save = model.module if hasattr(model, 'module') else model  # Only save the model it-self

    # If we save using the predefined names, we can load using `from_pretrained`
    output_model_file = os.path.join("./", "pytorch_model.bin")
    output_config_file = os.path.join("./", "bert_config.json")

    torch.save(model_to_save.state_dict(), output_model_file)
    model_to_save.config.to_json_file(output_config_file)
    tokenizer.save_vocabulary("./")
```

This gave my a a model .bin file and a config file.

The training accuracy was around 90% after the last epoch on 32.000 training samples, leaving 8.000 samples for evaluation. I then instantiated a new BERT model with from_pretrained method with state_dict as False and ran the evaluation which surprisingly gave these results:

{'eval_loss': 9.04939697444439, 'eval_accuracy': 0.036875}

I ran through the from_pretrained method and saw that the .bin file is a PyTorch dump of a BertForPreTraining instance which I presume means that the classifier weights are not saved when saving this way?

Then it would probably be necessary to pass the state_dict parameter when loading a classifier model with from_pretrained? If so, is it necessary to pass two files (.bin and state_dict as model_state) or is there another way to do this?

**Comment by thomwolf** [comment]
Did you try to load the model following the best-practices indicated here: https://github.com/huggingface/pytorch-pretrained-BERT#serialization-best-practices

**Comment by steindor** [comment]
All but load the tokenizer from the vocab file. Think that would make a large difference?
